src/fheroes2/maps/maps_tiles_helper.cpp

```diff
@@ -1166,6 +1167,57 @@ namespace
             }
         }
     }
+
+    void placeObjectOnTile( const Maps::Tiles & tile, const Maps::ObjectInfo & info )
+    {
+        assert( !info.empty() );
+
+        const uint32_t uid = Maps::getNewObjectUID();
+        const fheroes2::Point mainTilePos = tile.GetCenter();
+
+        for ( const auto & partInfo : info.groundLevelParts ) {
+            const fheroes2::Point pos = mainTilePos + partInfo.tileOffset;
+            if ( !Maps::isValidAbsPoint( pos.x, pos.y ) ) {
+                // This shouldn't happen as the object must be verified before placement.
+                assert( 0 );
+                continue;
+            }
+
+            const int32_t tileIndex = Maps::GetIndexFromAbsPoint( pos );
+            Maps::Tiles & currentTile = world.GetTiles( tileIndex );
```

Comment:
We can get tile by Point instead of calling `GetIndexFromAbsPoint()` and getting tile by index.
```suggestion
            Maps::Tiles & currentTile = world.GetTiles( pos.x, pos.y );
```